subprojects/core/src/test/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactoryTest.groovy

```diff
@@ -927,9 +928,13 @@ class AnnotationProcessingTaskFactoryTest extends AbstractProjectBuilderSpec imp
         T task = AbstractTask.injectIntoNewInstance(project, TaskIdentity.create(name, type, project), new Callable<T>() {
             T call() throws Exception {
                 if (params.length > 0) {
+                    // TODO: This should be using objectFactory too because it more closely matches
+                    // what the production code does. The production code is more lenient with types
+                    // and this just assumes the first constructor is the correct one.
+                    //  return TestUtil.objectFactory().newInstance(type, params)
```

Comment:
Why not do this now?

Reply:
I was trying to minimize the number of changes here. To properly fix this requires re-writing most of these tests because they rely on passing null to the constructor